addons/mail/models/mail_mail.py

```diff
@@ -107,6 +108,19 @@ def _inverse_unrestricted_attachment_ids(self):
             restricted_attaments = mail_sudo.attachment_ids - IrAttachment._filter_attachment_access(mail_sudo.attachment_ids.ids)
             mail_sudo.attachment_ids = restricted_attaments | mail.unrestricted_attachment_ids
 
+    def init(self):
+        """Create a partial index on "to_delete" to make the search on those records fast.
+
+        The benefit on this partial index is to not have a big impact on the
+        update / insert of other records in the database in comparison to a standard
+        index.
+        """
+        self._cr.execute("""
```

Comment:
Shouldn't this be done in migration script also ? Or does the module update create the index ?